### PR TITLE
animate title when waiting for input

### LINKED ISSUE #430: Feature: Update document title upon poll response from the backend
Suggestion: Update <title>Terraforming Mars</title> based on players' turns.

While api/waitingfor is polling:
- when you get {result: "WAIT", player: "Player 2"}, update the title to, e.g. "Player 2 is playing..."
- when you get {result: "GO", player: "Player 1"}, update the title to, e.g. "YOUR TURN"

Implementation is trivial with few lines of code.
https://developer.mozilla.org/en-US/docs/Web/API/Document/title

For better visualization when whose turn it is. Currently there is no notification if it is your turn if you dont keep the game as an active tab.

titles

### PR DESCRIPTION
**Pull request by bafolts**
This will close #430 . During multiplayer games animates the document title when user input is required. Most likely better animations. I selected `<^>v`. Suggestions welcome for better character sequences for the animation.

### CODE REVIEW

File: src/components/WaitingFor.ts [hunk 1]
```diff
@@ -64,6 +66,16 @@ export const WaitingFor = Vue.component('waiting-for', {
...
   methods: {
+    animateTitle: function() {
+      const sequence = '^>v<';
+      const first = document.title[0];
+      const position = sequence.indexOf(first);
+      let next = sequence[0];
+      if (position !== -1 && position < sequence.length - 1) {
+        next = sequence[position + 1];
+      }
+      document.title = next + ' ' + constants.APP_NAME;
```
Comment (kberg) [inline src/components/WaitingFor.ts:77]: Translateable?

**Review by kberg — APPROVED** [review]
Is this running somewhere?

**Comment by kberg** [comment]
Actually -- is this going to mean the title is always flashing in a solo game?

**Comment by vsrisuknimit** [comment]
@bafolt
1) ◴ ◷ ◶ ◵

**Comment by kberg** [comment]
@vsrisuknimit's alternatives are better.

File: src/components/WaitingFor.ts [hunk 2]
```diff
@@ -111,10 +124,15 @@ export const WaitingFor = Vue.component('waiting-for', {
     },
   },
   render: function(createElement) {
+    document.title = constants.APP_NAME;
+    window.clearInterval(documentTitleTimer);
     if (this.waitingfor === undefined) {
       this.waitForUpdate();
       return createElement('div', $t('Not your turn to take any actions'));
     }
+    if (this.player.players.length > 1 && this.player.waitingFor !== undefined) {
```
Comment (bafolts) [inline src/components/WaitingFor.ts:133]: @kberg this should ensure the title is only animated for games with more than one player. Solo games this wont animate.

Comment (kberg) [reply src/components/WaitingFor.ts:133]: Oh, of course. Excuse me for missing it.

**Comment by bafolts** [comment]
Used the clock which looks much better and added translation wrapper around the `document.title`. I don't think we translate the string `Terraforming Mars` today but will work if and when we do.

**Pull request merged**